src/integrationTest/java/org/opensearch/test/framework/matcher/GetSettingsResponseContainsIndicesMatcher.java

```diff
@@ -31,12 +34,16 @@ class GetSettingsResponseContainsIndicesMatcher extends TypeSafeDiagnosingMatche
 
 	@Override
 	protected boolean matchesSafely(GetSettingsResponse response, Description mismatchDescription) {
-		ImmutableOpenMap<String, Settings> indexToSettings = response.getIndexToSettings();
+
+		final Map<String, Settings> indexToSettings = new HashMap<>();
+		for (ObjectObjectCursor<String, Settings> cursor : response.getIndexToSettings()) {
+			indexToSettings.put(cursor.key, cursor.value);
+		}
```

Comment:
Can this be simplified to:

```suggestion
		final Set<String> indexKeys = new HashSet<>();
		for (ObjectObjectCursor<String, Settings> cursor : response.getIndexToSettings()) {
			indexKeys.add(cursor.key);
		}
```

Reply:
Looks like either would work, but since we are not using the values in the map and only getting a keySet it would take less memory.

Reply:
Sorry Craig, I didn't see this before the PR got merged. I had left it as a map because that is what the original implementation was. I agree that a Set should work fine here as well. If you wanted to swap it you could open up a quick PR to change it in main.